pilot/lib/pilot/tester_manager.rb

```diff
@@ -91,21 +95,35 @@ def find_app(app_filter: nil)
       nil
     end
 
-    def find_or_create_tester(email: nil, first_name: nil, last_name: nil)
-      tester = Spaceship::Tunes::Tester::Internal.find(config[:email])
-      tester ||= Spaceship::Tunes::Tester::External.find(config[:email])
+    def find_or_create_tester(email: nil, first_name: nil, last_name: nil, app: nil)
+      current_user = Spaceship::Members.find(Spaceship::Tunes.client.user)
+      tester = nil
+
+      if current_user.roles.include?("admin")
+        tester = Spaceship::Tunes::Tester::Internal.find(email)
+        tester ||= Spaceship::Tunes::Tester::External.find(email)
+      elsif current_user.roles.include?("appmanager")
+        raise "Account #{current_user.email_address} is only an 'App Manager' and therefore you must also define what app this tester (#{email}) should be added to" unless app
+        tester = Spaceship::Tunes::Tester::Internal.find_by_app(app.apple_id, email)
+        tester ||= Spaceship::Tunes::Tester::External.find_by_app(app.apple_id, email)
+      else
+        raise "Account #{current_user.email} doesn't have a role that is allowed to administer app testers, current roles: #{current_user.roles}"
+      end
 
       if tester
-        UI.success("Existing tester #{tester.email}")
-      else
-        tester = Spaceship::Tunes::Tester::External.create!(email: config[:email],
-                                                            first_name: config[:first_name],
-                                                            last_name: config[:last_name])
-        UI.success("Successfully added tester: #{tester.email} to your account")
+        UI.success("Existing tester #{email}")
+      elsif current_user.roles.include?("admin")
+        tester = Spaceship::Tunes::Tester::External.create!(email: email, first_name: first_name, last_name: last_name)
+        UI.success("Successfully added tester: #{email} to your account")
+      elsif current_user.roles.include?("appmanager")
+        Spaceship::TestFlight::Base.client.create_app_level_tester(app_id: app.apple_id, first_name: first_name, last_name: last_name, email: email)
```

Comment:
But this is all ugly because we didnt make a clean cut over from the Tunes namespaced testers because those are still represented in the Users and Roles view of iTC